packages/aws-cdk-lib/aws-iam/lib/role.ts

```diff
@@ -413,6 +413,10 @@ export class Role extends Resource implements IRole {
       physicalName: props.roleName,
     });
 
+    if (props.roleName && !/^[\w_+=,.@-]{1,64}$/.test(props.roleName)) {
```

Comment:
We need to cover the case where the roleName is a token, and thus we cannot parse it and see if the roleName is valid. We let these go through to CloudFormation, and there's tons of prior art around CDK that handles these situations. Need a unit test for it as well.

Reply:
I'm going to add this check and relative tests